neurom/fst/sectionfunc.py

```diff
@@ -91,3 +91,35 @@ def section_meander_angles(section):
     p = section.points
     return [mm.angle_3points(p[i - 1], p[i - 2], p[i])
             for i in range(2, len(p))]
+
+
+def strahler_order(section):
+    '''Branching order of a tree section
+
+    The strahler order is the inverse of the branch order,
+    since this is computed from the tips of the tree
+    towards the root.
+
+    This implementation is a translation of the three steps described in
+    Wikipedia (https://en.wikipedia.org/wiki/Strahler_number):
+     - If the node is a leaf (has no children), its Strahler number is one.
+     - If the node has one child with Strahler number i, and all other children
+       have Strahler numbers less than i, then the Strahler number of the node
+       is i again.
+     - If the node has two or more children with Strahler number i, and no
+       children with greater number, then the Strahler number of the node is
+       i + 1.
+
+    No efforts have been invested in making it computationnaly efficient, but
+    it computes acceptably fast on tested morphologies (i.e., no waiting time).
+    '''
+    if section.children:
+        child_orders = [strahler_order(child) for child in section.children]
+        max_so_children = max(child_orders)
+        it = iter(co == max_so_children for co in child_orders)
```

Comment:
n_max_so_children = sum(co == max_so_children for co in child_orders)
seems more pythonic to me but that's a detail

Reply:
that requires going through the list fully, as opposed to this way which only goes as far as finding 2 which match; probably a pointless optimization